File: php/features.php
<?php
declare(strict_types=1);

// Gbif SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GbifFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GbifBaseFeature();
            case "test":
                return new GbifTestFeature();
            default:
                return new GbifBaseFeature();
        }
    }
}

File: py/features.py
# Gbif SDK feature factory

from feature.base_feature import GbifBaseFeature
from feature.test_feature import GbifTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GbifBaseFeature(),
        "test": lambda: GbifTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
